enacts/onset/agronomy.py

```diff
@@ -73,38 +76,95 @@ def soil_plant_water_balance(
         total available water that represents the maximum water capacity of the soil.
     sminit : DataArray
         timeless soil moisture to initialize the loop with.
+    kc_params : DataArray
+        Crop Cultivar Kc parameters as a function of the inflexion points of the Kc curve,
+        expressed in consecutive daily time deltas originating from `planting_date`
+        as coordinate `kc_periods` (default `kc_params` =None in which case Kc is set to 1).
+    planting_date : DataArray
+        dates when planting (default `planting_date` =None -- not covered yet)
     time_dim : str, optional
         daily time dimension to run the balance against (default `time_dim` ="T").
         
     Returns
     -------
-    sm, drainage : Tuple of DataArray
-        daily soil moisture and drainage over the growing season.
+    sm, drainage, et_crop : Tuple of DataArray
+        daily soil moisture, drainage and crop evapotranspiration over the growing season.
         
     See Also
     --------
     soil_plant_water_step
     
+    Notes
+    -----
+    The daily evapotranspiration `et` can be scaled by a Crop Cultivar Kc
+    modelizing a crop needs in water according to the stage of its growth.
+    Kc is set to 1 outside of the growing period. i.e. before planting date
+    and after the last Kc curve inflexion point.
+    
+    Examples
+    --------
+    Example of kc_params:
+    
+    >>> kc_periods = pd.TimedeltaIndex([0, 45, 47, 45, 45], unit="D")
+    >>> kc_params = xr.DataArray(
+    >>>    data=[0.2, 0.4, 1.2, 1.2, 0.6], dims=["kc_periods"], coords=[kc_periods]
+    >>> )
+    <xarray.DataArray (kc_periods: 5)>
+    array([0.2, 0.4, 1.2, 1.2, 0.6])
+    Coordinates:
+        * kc_periods  (kc_periods) timedelta64[ns] 0 days 45 days ... 45 days 45 days
+    
+    Example of planting_date:
+    
+    >>> p_d = xr.DataArray(
+    >>>    pd.DatetimeIndex(data=["2000-05-02", "2000-05-13"]),
```

Comment:
This is arbitrary indeed (other than that it is a valid date for the precip_sample used in tests).

This also likely temporary. Right now I am covering the case where we pretend we care for only one year at a time and I don't have the vision of how things will change when we deal with multiple years.

2 variables might not be great because one, a user could use for instance the ouput of onset_date as the input for planting_date (Well I guess one could get onset date and make it 2 variables)...

So either we think carefully of the multiple year now. Either we let that go until we do.

Reply:
Ok, we can leave it for now.